src/bvals/boundary_conditions_generic.hpp

```diff
@@ -138,7 +138,7 @@ void GenericSwarmBC(std::shared_ptr<Swarm> &swarm) {
 }
 
 namespace impl {
-using desc_key_t = std::tuple<bool, TopologicalType>;
+using desc_key_t = std::tuple<bool, bool, TopologicalType>;
```

Comment:
Not suggesting for this PR, but if this gets further expanded it might be worth to introduce enum classes or structs to be passed as parameter (to reduce a potential `{true, false, true, ...}` confusion)

Reply:
Agreed, but I also agree we should leave for a future PR.